include/rocksdb/env.h

```diff
@@ -535,6 +535,11 @@ class WritableFile {
     }
   }
 
+  // Indicates the upper layers if the current WritableFile implementation
+  // supports direct IO.
+  bool SupportsDirectIO() {
```

Comment:
I did not make it virtual to disable direct IO by default. I can do so and implement in posix and HDFS backends.
Regarding the name, direct IO is supported by the backend implementing WritableFile, so it does not go all the way up to user options (though it might be in the future).